staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/meta.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategicpatch
+
+import (
+	"reflect"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/mergepatch"
+	forkedjson "k8s.io/apimachinery/third_party/forked/golang/json"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/util/openapi"
+)
+
+type PatchMeta struct {
+	PatchStrategies []string
+	PatchMergeKey   string
+}
+
+type LookupPatchMeta interface {
```

Comment:
Could you document that?

It's very unclear what `Elem()` does

Reply:
It is for array, since struct will need to get the subtype.
```
func (s PatchMetaFromStruct) Elem() LookupPatchMeta {
	return PatchMetaFromStruct{T: s.T.Elem()}
}
```